pyproject.toml

```diff
@@ -18,8 +18,8 @@ dependencies = [
 
   # --------- basics --------- #
   "pyrootutils ~= 1.0.4",      # standardizing the project root setup
-  "rich ~= 12.6.0",            # beautiful text formatting in terminal
-  "fire == 0.5.0",
+  "rich",                      # beautiful text formatting in terminal
+  "fire",                      # automated CLI
```

Comment:
I would encourage you to lock these dependencies to at least some major version and ideally some minor version too.

Reply:
For rich: https://github.com/Lightning-AI/pytorch-lightning/blob/cd8acc26c3dd4d37a2b9aca458bd0d511d77df5c/requirements/app/app.txt#L25

Reply:
My reason not to lock them is the potential conflict with other packages. For example, Anomalib requires "rich>=13.5.2", and we may see conflict constraints on rich/fire in the future application of MART.

Maybe we should put rich and fire into an optional dependency group, so that future users can easily bypass the version constraints? Maybe we should not have "required" dependency at all?

Reply:
So I would tag rich as "rich >= 13.5.2, < 13.6.0". Having deps that fail loudly in this case is good. Fire is totally controlled by us since unless some other dependency relies upon it?

Reply:
OK. I have locked both in [9d3a471](https://github.com/IntelLabs/MART/pull/252/commits/9d3a471b98b66f8a59feb0fd93b3d6e09b9a5c15)